app/lib/dfe/wizard.rb

```diff
@@ -0,0 +1,138 @@
+module DfE
+  class Wizard
+    attr_reader :current_step_name, :steps, :request
+    attr_writer :step_params
+
+    delegate :next_step, to: :current_step
+    delegate :info, to: :logger, allow_nil: true
+
+    def initialize(current_step:, step_params: {}, **args)
+      @current_step_name = current_step
+      @step_params = step_params
+      @steps = self.class.steps
+
+      (args || {}).each do |key, value|
+        send("#{key}=", value) if respond_to?("#{key}=")
+      end
+    end
+
+    def self.steps
+      return @steps unless block_given?
```

Comment:
Is there a test example where there is no block passed to the `.steps` method?

Reply:
When we initialize the wizards I pass steps so it can be an instance variable of the wizard via `@wizard.steps`